components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/autofill/GeckoLoginDelegateWrapper.kt

```diff
@@ -14,18 +16,26 @@ import org.mozilla.geckoview.LoginStorage
  * them to [storageDelegate]. This allows us to avoid duplicating [LoginStorageDelegate] code
  * between different versions of GeckoView, by duplicating this wrapper instead.
  */
-class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) : LoginStorage.Delegate {
+class GeckoLoginDelegateWrapper(private val storageDelegate: LoginStorageDelegate) :
+    LoginStorage.Delegate {
     override fun onLoginSave(login: LoginStorage.LoginEntry) {
         storageDelegate.onLoginSave(login.toLogin())
     }
 
     override fun onLoginFetch(domain: String): GeckoResult<Array<LoginStorage.LoginEntry>>? {
-        val storedLogins = storageDelegate.onLoginFetch(domain)
+        val result = GeckoResult<Array<LoginStorage.LoginEntry>>()
 
-        return storedLogins
-            .map { it.toLoginEntry() }
-            .toTypedArray()
-            .let { GeckoResult.fromValue(it) }
+        GlobalScope.launch {
+            val storedLogins = storageDelegate.onLoginFetch(domain)
+
```

Comment:
I remember we tested some edge cases for this, but I can't remember exactly what.  The following are the ones that come to mind.

- Scope is launched -> user backs to home fragment (this one I know we checked
- Scope is launched -> user navigates to a new page
- Scope is launched -> user backgrounds the app
- Scope is launched -> user refreshes the page

If all of these are OK, then I'm fine merging this.

Reply:
I think all of these are fine, but we should probably also explicitly define a dispatcher?

Reply:
Yeah, I think with us using the storage here it makes sense to explicitly use the IO dispatcher.